pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -43,7 +43,19 @@ type TaskRunSpec struct {
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
 	// +optional
-	ServiceAccount string `json:"serviceAccount"`
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+	// +optional
+	PVCName string `json:"pvcName,omitempty"`
+	// +optional
+	PreBuiltSteps []TaskBuildStep `json:"preBuildSteps,omitempty"`
+	// +optional
+	PostBuiltSteps []TaskBuildStep `json:"postBuiltSteps,omitempty"`
```

Comment:
I leaned toward `Built ` because thats the end object we are creating now. I like the name `BeforeSteps` for pre build steps, for post build steps we could go with `AfterSteps`.